Makefile

```diff
@@ -47,123 +42,111 @@ $(GOBIN):
 
 work: $(GOBIN)
 
-depend: work
```

Comment:
why depend is removed? Imo we should have way to update components like "previously" see my PR https://github.com/kubernetes/cloud-provider-openstack/pull/816 about depend things

Reply:
@kayrus removing depend causing changes at many place, @zetaab in your PR too depend-update is not doing the same thing, for go modules i don't think there is a requirement for separate update command

Reply:
@kayrus with your PR we not creating any vendor directory before build? i see there is no requirement but @zetaab you are doing this any particular reason for that?

Reply:
@adisky there is no requirement in using the vendor dir, except the `import-boss`  for tests. more details are here: https://github.com/golang/go/wiki/Modules#can-i-control-when-gomod-gets-updated-and-when-the-go-tools-use-the-network-to-satisfy-dependencies

Reply:
I did vendor folder because that is how I have always done things. However, it is same to me is there vendor folder or not. The main thing is to get this PR forward. :)

Reply:
@zetaab I am ok with the changes, let the test pass and also fine in both ways having depend or not.